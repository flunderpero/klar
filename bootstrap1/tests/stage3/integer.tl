let num = 40
let ok = "OK: to_string() == "
let n = "variable name should not clash with `num`"
let s = ""

-- Addition
print("Addition")
num = num + 2
s = num.to_string()
print(s)

-- Subtraction
print("Subtraction")
num = num - 2
s = num.to_string()
print(s)

if num == 42 then
    print("OK: `num` is 42")
end

if num == 99 then
    print("ERR: `num` is not 99")
end

-- Stringify
ok = ok + num.to_string()
print(ok)
ok = num.to_string()
print(ok)

-- Test for variable name clash.
n = "OK: no variable name clash"
print(n)
