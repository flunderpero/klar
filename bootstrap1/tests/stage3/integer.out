Addition
42
Subtraction
40
OK: to_string() == 40
40
OK: no variable name clash
